### PR TITLE
[Add] Remove the default price list in quotation/sale

### PR DESCRIPTION
**Pull request by ehdem**
In quotaion/sale form view, display the configured price list
for the selected partner

### CODE REVIEW

File: sale_remove_default_pricelist/sale.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+
+class sale(orm.Model):
+    _inherit = 'sale.order'
+
+    def onchange_shop_id(self, cr, uid, ids, shop_id, context=None):
+        """Copied form the original function. Remove the default
+        value for price list. Price list can be configured for each partner."""
+        context = context or {}
+        v = super(sale, self).onchange_shop_id(
```
Comment (joaoalf) [inline sale_remove_default_pricelist/sale.py]: @ehdem It's better to receive the whole return dictionary (not just the 'value' one). If another method in the inherit chain changes a domain or adds a message we will miss it.

Comment (ehdem) [inline sale_remove_default_pricelist/sale.py]: @joaoalf : Ok

File: sale_remove_default_pricelist/sale.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    def onchange_shop_id(self, cr, uid, ids, shop_id, context=None):
+        """Copied form the original function. Remove the default
+        value for price list. Price list can be configured for each partner."""
+        context = context or {}
+        res = super(sale, self).onchange_shop_id(
+            cr, uid, ids, shop_id, context=context)
+        if 'pricelist_id' in res['value']:
+            del res['value']['pricelist_id']
+        return {'value': res['value']}
```
Comment (joaoalf) [inline sale_remove_default_pricelist/sale.py]: @ehdem Why not return res?

Comment (ehdem) [inline sale_remove_default_pricelist/sale.py]: @joaoalf : Ok

File: sale_remove_default_pricelist/sale.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    def onchange_shop_id(self, cr, uid, ids, shop_id, context=None):
+        """Copied form the original function. Remove the default
+        value for price list. Price list can be configured for each partner."""
+        context = context or {}
+        res = super(sale, self).onchange_shop_id(
+            cr, uid, ids, shop_id, context=context)
+        if 'pricelist_id' in res['value']:
+            del res['value']['pricelist_id']
+        return {'value': res}
```
Comment (joaoalf) [inline sale_remove_default_pricelist/sale.py]: @ehdem res has already a value key. Why not return directly res. Ex:

``` python
return res
```

**Comment by joaoalf** [comment]
LGTM
:+1:

**Comment by jgrandguillaume** [comment]
Hi,

Don't know why runbot is red, but otherwise :+1:

**Comment by ehdem** [comment]
@jgrandguillaume: I think it is an error in the log when trying to install 'sale_line_price_properties_based' module

**Comment by ehdem** [comment]
Can't access the from https://runbot.odoo-community.org/runbot/build/2925951

**Comment by max3903** [comment]
:+1:

**Pull request merged**